In [32]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn import metrics
plt.style.use("ggplot")
%matplotlib inline

In [39]:
data = pd.read_csv('corona_world.csv')
# Removing the sno column
data = data.drop(['Sno'], axis = 1)
# data['Country'].head(50)

Date    Province/State               Country      Last Update  \
720  2/4/2020 22:00          Liaoning        Mainland China   4/2/2020 14:13   
721  2/4/2020 22:00            Shanxi        Mainland China   4/2/2020 23:23   
722  2/4/2020 22:00           Tianjin        Mainland China   4/2/2020 13:33   
723  2/4/2020 22:00           Guizhou        Mainland China    5/2/2020 1:23   
724  2/4/2020 22:00             Gansu        Mainland China   4/2/2020 15:43   
725  2/4/2020 22:00             Jilin        Mainland China    5/2/2020 0:43   
726  2/4/2020 22:00    Inner Mongolia        Mainland China    5/2/2020 1:53   
727  2/4/2020 22:00           Ningxia        Mainland China    4/2/2020 2:13   
728  2/4/2020 22:00          Xinjiang        Mainland China    5/2/2020 2:03   
729  2/4/2020 22:00               NaN              Thailand   4/2/2020 15:33   
730  2/4/2020 22:00               NaN             Singapore   4/2/2020 15:33   
731  2/4/2020 22:00               NaN                 Japan   4/2/2020 16:43   
732  2/4/2020 22:00         Hong Kong             Hong Kong    5/2/2020 2:33   
733  2/4/2020 22:00           Qinghai        Mainland China    5/2/2020 1:23   
734  2/4/2020 22:00               NaN           South Korea   4/2/2020 15:33   
735  2/4/2020 22:00               NaN               Germany   3/2/2020 20:53   
736  2/4/2020 22:00            Taiwan                Taiwan   4/2/2020 14:53   
737  2/4/2020 22:00             Macau                 Macau    4/2/2020 4:43   
738  2/4/2020 22:00               NaN              Malaysia    4/2/2020 7:33   
739  2/4/2020 22:00               NaN               Vietnam   3/2/2020 21:43   
740  2/4/2020 22:00               NaN                France    1/2/2020 1:52   
741  2/4/2020 22:00               NaN  United Arab Emirates    2/2/2020 5:43   
742  2/4/2020 22:00   New South Wales             Australia   1/2/2020 18:12   
743  2/4/2020 22:00          Victoria             Australia   1/2/2020 18:12   
744  2/4/2020 22:00        Queensland             Australia   4/2/2020 16:53   
745  2/4/2020 22:00               NaN                 India   3/2/2020 21:43   
746  2/4/2020 22:00   South Australia             Australia   2/2/2020 22:33   
747  2/4/2020 22:00       Toronto, ON                Canada    4/2/2020 0:13   
748  2/4/2020 22:00               NaN                 Italy   1/31/2020 8:15   
749  2/4/2020 22:00               NaN           Philippines    2/2/2020 3:33   
750  2/4/2020 22:00               NaN                Russia  1/31/2020 16:13   
751  2/4/2020 22:00               NaN                    UK    1/2/2020 1:52   
752  2/4/2020 22:00       Chicago, IL                    US   1/2/2020 19:43   
753  2/4/2020 22:00    San Benito, CA                    US    3/2/2020 3:53   
754  2/4/2020 22:00   Santa Clara, CA                    US    3/2/2020 0:43   
755  2/4/2020 22:00               NaN               Belgium   4/2/2020 15:43   
756  2/4/2020 22:00               NaN              Cambodia   1/31/2020 8:15   
757  2/4/2020 22:00  British Columbia                Canada   1/2/2020 18:12   
758  2/4/2020 22:00        London, ON                Canada    4/2/2020 0:03   
759  2/4/2020 22:00               NaN               Finland   1/31/2020 8:15   
760  2/4/2020 22:00             Tibet        Mainland China    1/2/2020 1:52   
761  2/4/2020 22:00               NaN                 Nepal   1/31/2020 8:15   
762  2/4/2020 22:00               NaN                 Spain   1/2/2020 23:43   
763  2/4/2020 22:00               NaN             Sri Lanka   1/31/2020 8:15   
764  2/4/2020 22:00               NaN                Sweden    1/2/2020 2:13   
765  2/4/2020 22:00        Boston, MA                    US   1/2/2020 19:43   
766  2/4/2020 22:00   Los Angeles, CA                    US   1/2/2020 19:53   
767  2/4/2020 22:00        Orange, CA                    US   1/2/2020 19:53   
768  2/4/2020 22:00       Seattle, WA                    US   1/2/2020 19:43   
769  2/4/2020 2

In [47]:
# converting to datetime
data['Date'] = pd.to_datetime(data['Date'])

## COMBINING CHINA AND MAINLAND CHINA CASES 

In [48]:
data = data.replace(to_replace = ['Mainland China'], value = ['China'])

data.tail(50)

Date    Province/State               Country  \
720 2020-02-04 22:00:00          Liaoning                 China   
721 2020-02-04 22:00:00            Shanxi                 China   
722 2020-02-04 22:00:00           Tianjin                 China   
723 2020-02-04 22:00:00           Guizhou                 China   
724 2020-02-04 22:00:00             Gansu                 China   
725 2020-02-04 22:00:00             Jilin                 China   
726 2020-02-04 22:00:00    Inner Mongolia                 China   
727 2020-02-04 22:00:00           Ningxia                 China   
728 2020-02-04 22:00:00          Xinjiang                 China   
729 2020-02-04 22:00:00               NaN              Thailand   
730 2020-02-04 22:00:00               NaN             Singapore   
731 2020-02-04 22:00:00               NaN                 Japan   
732 2020-02-04 22:00:00         Hong Kong             Hong Kong   
733 2020-02-04 22:00:00           Qinghai                 China   
734 2020-02-04 22:00:00               NaN           South Korea   
735 2020-02-04 22:00:00               NaN               Germany   
736 2020-02-04 22:00:00            Taiwan                Taiwan   
737 2020-02-04 22:00:00             Macau                 Macau   
738 2020-02-04 22:00:00               NaN              Malaysia   
739 2020-02-04 22:00:00               NaN               Vietnam   
740 2020-02-04 22:00:00               NaN                France   
741 2020-02-04 22:00:00               NaN  United Arab Emirates   
742 2020-02-04 22:00:00   New South Wales             Australia   
743 2020-02-04 22:00:00          Victoria             Australia   
744 2020-02-04 22:00:00        Queensland             Australia   
745 2020-02-04 22:00:00               NaN                 India   
746 2020-02-04 22:00:00   South Australia             Australia   
747 2020-02-04 22:00:00       Toronto, ON                Canada   
748 2020-02-04 22:00:00               NaN                 Italy   
749 2020-02-04 22:00:00               NaN           Philippines   
750 2020-02-04 22:00:00               NaN                Russia   
751 2020-02-04 22:00:00               NaN                    UK   
752 2020-02-04 22:00:00       Chicago, IL                    US   
753 2020-02-04 22:00:00    San Benito, CA                    US   
754 2020-02-04 22:00:00   Santa Clara, CA                    US   
755 2020-02-04 22:00:00               NaN               Belgium   
756 2020-02-04 22:00:00               NaN              Cambodia   
757 2020-02-04 22:00:00  British Columbia                Canada   
758 2020-02-04 22:00:00        London, ON                Canada   
759 2020-02-04 22:00:00               NaN               Finland   
760 2020-02-04 22:00:00             Tibet                 China   
761 2020-02-04 22:00:00               NaN                 Nepal   
762 2020-02-04 22:00:00               NaN                 Spain   
763 2020-02-04 22:00:00               NaN             Sri Lanka   
764 2020-02-04 22:00:00               NaN                Sweden   
765 2020-02-04 22:00:00        Boston, MA                    US   
766 2020-02-04 22:00:00   Los Angeles, CA                    US   
767 2020-02-04 22:00:00        Orange, CA                    US   
768 2020-02-04 22:00:00       Seattle, WA                    US   
769 2020-02-04 22:00:00         Tempe, AZ                    US   

         Last Update  Confirmed  Deaths  Recovered  
720   4/2/2020 14:13         81       0          2  
721   4/2/2020 23:23         81       0          4  
722   4/2/2020 13:33         67       0          2  
723    5/2/2020 1:23         64       0          5  
724   4/2/2020 15:43         57       0          4  
725    5/2/2020 0:43         54       0          1  
726    5/2/2020 1:53         42       0          1  
727    4/2/2020 2:13         34       0          1  
728    5/2/2020 2:03         32       0          0  
729   4/2/2020 15:33         25       0          5  
730   4/2/2020 15:

- DATA SHOWS THERE ARE A TOTAL OF 770 DATA IN THE COLLECTION WITH DATE,COUNTRY, LAST UPDATE, CONFIRMED, DEATHS, RECOVERED AND PROVINCE/STATE BEING THE ONLY COLUMNS WITH MISSING OR NULL VALUES

### CALCULATING THE TOTAL COUNTRIES AFFECTED AFTER THE COMBINATION

In [58]:
total_countries = data['Country'].unique().tolist()
no = len(total_countries)
no

31

- After combining China and MainLand China, the total number of country affect is 31

In [60]:
data.describe()

Confirmed      Deaths   Recovered
count    770.000000  770.000000  770.000000
mean     160.207792    3.436364    4.264935
std     1015.504102   31.553727   28.718715
min        0.000000    0.000000    0.000000
25%        2.000000    0.000000    0.000000
50%        8.000000    0.000000    0.000000
75%       58.750000    0.000000    1.000000
max    16678.000000  479.000000  522.000000

In [37]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 770 entries, 0 to 769
Data columns (total 7 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Date            770 non-null    datetime64[ns]
 1   Province/State  585 non-null    object        
 2   Country         770 non-null    object        
 3   Last Update     770 non-null    object        
 4   Confirmed       770 non-null    int64         
 5   Deaths          770 non-null    int64         
 6   Recovered       770 non-null    int64         
dtypes: datetime64[ns](1), int64(3), object(3)
memory usage: 33.1+ KB


- AFTER SOME RESEARCHES THE INTERNET SHOWS THAT THERE WERE A TOTAL OF 2,817 DEATH COUNT AS AT 27/02/2020.WHLE THIS DATA SHOW JUST 770 TAKEN BETWEEN 22/01/2020 TO 04/02/2020. THIS DATASET MIGHT NOT BE VERY ACCURATE DUE TO INCOMPLETE DATA 

- ALSO ON AN AVERAGE 160 OUT OF 770 COUNTS HAS BEEN CONFIRMED WITH THE VIRUS

In [14]:
data.corr()

Sno  Confirmed    Deaths  Recovered
Sno        1.000000   0.083996  0.052086   0.104204
Confirmed  0.083996   1.000000  0.988916   0.986025
Deaths     0.052086   0.988916  1.000000   0.963501
Recovered  0.104204   0.986025  0.963501   1.000000

Date   Province/State Country      Last Update  Confirmed  \
0   2020-01-22 12:00:00            Anhui   China  1/22/2020 12:00          1   
1   2020-01-22 12:00:00          Beijing   China  1/22/2020 12:00         14   
2   2020-01-22 12:00:00        Chongqing   China  1/22/2020 12:00          6   
3   2020-01-22 12:00:00           Fujian   China  1/22/2020 12:00          1   
4   2020-01-22 12:00:00            Gansu   China  1/22/2020 12:00          0   
..                  ...              ...     ...              ...        ...   
765 2020-02-04 22:00:00       Boston, MA      US   1/2/2020 19:43          1   
766 2020-02-04 22:00:00  Los Angeles, CA      US   1/2/2020 19:53          1   
767 2020-02-04 22:00:00       Orange, CA      US   1/2/2020 19:53          1   
768 2020-02-04 22:00:00      Seattle, WA      US   1/2/2020 19:43          1   
769 2020-02-04 22:00:00        Tempe, AZ      US   1/2/2020 19:43          1   

     Deaths  Recovered  
0         0          0  
1         0          0  
2         0          0  
3         0          0  
4         0          0  
..      ...        ...  
765       0          0  
766       0          0  
767       0          0  
768       0          0  
769       0          0  

[770 rows x 7 columns]

In [ ]:
plt.scatter(x=df['Confirmed'], 
y=df['Confirmed'], 
s=df['Population']/1000, 
alpha=0.5, 
c=df['Bubble Color'])